en/docs/learn/api-controller/advanced-topics/configuring-environment-specific-parameters.md

```diff
@@ -0,0 +1,82 @@
+#  Configuring Environment Specific Parameters
+
+When there are multiple environments, to allow easily configuring environment-specific details, apictl supports an additional parameter file named `api_params.yaml`. It is recommended to store the parameter file with the API Project; however, it can be stored anywhere as required. 
+
+After the file is placed in the project directory, the tool will auto-detect the parameters file upon running the `import-api` command and create an environment-based artifact for API Manager. If the `api_params.yaml` is not found in the project directory, the tool will lookup in the project’s base path and the current working directory. 
+
+The following is the structure of the parameter file.
+
+```go
+environments:
+    - name: <environment_name>
+      endpoints:
+          production:
+              url: <production_endpoint_url>
+              config:
+                  retryTimeOut: <no_of_retries_before_suspension>
+                  retryDelay: <retry_delay_in_ms>
+                  factor: <suspension_factor>
+          sandbox:
+              url: <sandbox_endpoint_url>
+              config:
+                  retryTimeOut: <no_of_retries_before_suspension>
+                  retryDelay: <retry_delay_in_ms>
+                  factor: <suspension_factor>
+      security:
+        - enabled: <whether_security_is_enabled>
+          type: <endpoint_authentication_type_basic_or_digest>
+          username: <endpoint_username>
+          password: <endpoint_password>
+      gatewayEnvironments:
+        - <gateway_environment_name>           
+      certs:
+        - hostName: <endpoint_url>
+          alias: <certificate_alias>
+          path: <certificate_file_path>
+```
+The following code snippet contains sample configuration of the parameter file.
+
+!!! example
+    ```go
+    environments:
+        - name: dev
+          endpoints:
+            production:
+              url: 'https://dev.wso2.com'
+          security:
+           - enabled: true
+             type: basic
+             username: admin
+             password: admin
+          certs:
+            - hostName: 'https://dev.wso2.com'
+              alias: Dev
+              path: ~/.certs/dev.pem 
+          gatewayEnvironments:
+            - Production and Sandbox    
+        - name: test
+          endpoints:
+            production:
+              url: 'https://test.wso2.com'
+          security:
+           - enabled: true
+             type: digest
+             username: admin
+             password: admin
+              config:
+                retryTimeOut: $RETRY
+            sandbox:
+              url: 'https://test.sandbox.wso2.com'
+    ```
+Instead of the default `api_params.yaml`, you can a provide custom parameter file using `--params` flag. A sample command will be as follows.
```

Comment:
Shall we change this to "you can provide a"?

Reply:
Fixed via https://github.com/wso2/docs-apim/pull/3097/commits/74cdaa1323d12a99abd038f41a690914b8d50ab5

Reply:
Fixed via https://github.com/wso2/docs-apim/pull/3098/commits/edd8ae4c020155b3c9d8e4465a923a031e8f9472 for 3.2.0

Reply:
Fixed via https://github.com/wso2/docs-apim/pull/3099/commits/5f5c672fdf5973c1368873c40f65a138e2fbe58a for 3.1.0

Reply:
Fixed via https://github.com/wso2/docs-apim/pull/3100/commits/0dde1cd55e1981bc950fada7f2b96ba5263d5deb for 3.0.0